src/DataStructures/CompressedVector.hpp

```diff
@@ -0,0 +1,64 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// `blaze::CompressedVector` is a general-purpose sparse vector type. This file
+/// implements interoperability of `blaze::CompressedVector` with our data
+/// structures.
+
+#pragma once
+
+#include <blaze/math/CompressedVector.h>
+#include <cstddef>
+#include <pup.h>
+#include <vector>
+
+#include "Options/ParseOptions.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace PUP {
+/// @{
+/// Serialization of blaze::CompressedVector
+template <typename T, bool TF, typename Tag>
+void pup(er& p, blaze::CompressedVector<T, TF, Tag>& t) {
+  size_t size = t.size();
+  p | size;
+  size_t num_non_zeros = t.nonZeros();
+  p | num_non_zeros;
+  size_t index;
+  if (p.isUnpacking()) {
+    t.resize(size);
+    for (size_t i = 0; i < num_non_zeros; ++i) {
+      p | index;
+      p | t[index];
+    }
+  } else {
+    for (auto it = t.begin(); it != t.end(); ++it) {
+      index = it->index();
+      p | index;
+      p | it->value();
+    }
+  }
+}
+template <typename T, bool TF, typename Tag>
+void operator|(er& p, blaze::CompressedVector<T, TF, Tag>& t) {
+  pup(p, t);
+}
+/// @}
+}  // namespace PUP
+
+template <typename T, bool TF, typename Tag>
+struct Options::create_from_yaml<blaze::CompressedVector<T, TF, Tag>> {
+  template <typename Metavariables>
+  static blaze::CompressedVector<T, TF, Tag> create(
+      const Options::Option& options) {
+    const auto data = options.parse_as<std::vector<T>>();
+    blaze::CompressedVector<T, TF, Tag> result(data.size());
+    for (size_t i = 0; i < data.size(); ++i) {
+      if (gsl::at(data, i) != 0.) {
```

Comment:
could you add a comment why not just use `std::copy` (I understand why, but better to communicate to sleepy Nils in the future :) )